### PR TITLE
[xtext-xtend#397][WIP] Xbase EPackage changes to prepare ternary operator and try with resource

### PR DESCRIPTION
**Pull request by szarnekow**
To be fully backwards compatible, we would need to introduce a new EPackage that extends the Xbase package. @cdietrich What is your take on this?

### CODE REVIEW

File: org.eclipse.xtext.xbase/model/Xbase.ecore [hunk 1]
```diff
@@ -277,7 +279,7 @@
         containment="true"/>
     <eStructuralFeatures xsi:type="ecore:EReference" name="eachExpression" eType="#//XExpression"
         containment="true"/>
-    <eStructuralFeatures xsi:type="ecore:EReference" name="declaredParam" eType="ecore:EClass platform:/resource/org.eclipse.xtext.common.types/model/JavaVMTypes.ecore#//JvmFormalParameter"
+    <eStructuralFeatures xsi:type="ecore:EReference" name="declaredParam" eType="ecore:EClass ../../org.eclipse.xtext.common.types/model/JavaVMTypes.ecore#//JvmFormalParameter"
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/model/Xbase.ecore]: is this change intended?

Comment (szarnekow) [reply org.eclipse.xtext.xbase/model/Xbase.ecore]: Looks like a new EMF version did some fance optimization. Good catch.

File: org.eclipse.xtext.xbase/emf-gen/org/eclipse/xtext/xbase/XbasePackage.java [hunk 2]
```diff
@@ -1901,6 +1919,17 @@
 	 */
 	EReference getXIfExpression_Else();
 
+	/**
+	 * Returns the meta object for the attribute '{@link org.eclipse.xtext.xbase.XIfExpression#isConditionalExpression <em>Conditional Expression</em>}'.
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @return the meta object for the attribute '<em>Conditional Expression</em>'.
+	 * @see org.eclipse.xtext.xbase.XIfExpression#isConditionalExpression()
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/emf-gen/org/eclipse/xtext/xbase/XbasePackage.java]: what is isConditionalExpression good for?

Comment (cdietrich) [reply org.eclipse.xtext.xbase/emf-gen/org/eclipse/xtext/xbase/XbasePackage.java]: is that ?: ?

File: org.eclipse.xtext.xbase/plugin.properties [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2011 itemis AG (http://www.itemis.eu) and others.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+
+pluginName = Xbase Model
+providerName = www.example.org
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase/plugin.properties]: looks bad to me

**Comment by cdietrich** [comment]
what is exactly not full backwards compatible?
or is that a binary thing
(subclass with multi inheritance?)

File: org.eclipse.xtext.xbase/emf-gen/org/eclipse/xtext/xbase/XbasePackage.java (hunk 2 above)
Comment (szarnekow) [reply org.eclipse.xtext.xbase/emf-gen/org/eclipse/xtext/xbase/XbasePackage.java]: It allows to detect if we had a ternary expression or an if expression. This will sport warnings and quickfixes.

**Comment by szarnekow** [comment]
> what is exactly not full backwards compatible?

The API of XbasePackage and XbasePackage.Literals, XTryCatchFinallyExpression and XIfExpression is extended.

**Comment by cdietrich** [comment]
please fix the failing test
http://services.typefox.io/open-source/jenkins/job/xtext-extras/job/sz_issue397/lastCompletedBuild/testReport/

File: org.eclipse.xtext.xbase.ide/src-gen/org/eclipse/xtext/xbase/annotations/ide/contentassist/antlr/internal/InternalXbaseWithAnnotationsParser.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
 import java.util.Map;
 import java.util.HashMap;
 /*******************************************************************************
- * Copyright (c) 2010-2017 itemis AG (http://www.itemis.eu) and others.
+ * Copyright (c) 2010, 2018 itemis AG (http://www.itemis.eu) and others.
```
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.xbase.ide/src-gen/org/eclipse/xtext/xbase/annotations/ide/contentassist/antlr/internal/InternalXbaseWithAnnotationsParser.java]: Can you open a bug for that

**Comment by szarnekow** [comment]
> is that ?: ?

It's a boolean flag that allows to detect an XIfExpression of the form `condition ? thenExpr : elseExpr`

**Comment by szarnekow** [comment]
> please fix the failing test
done

File: org.eclipse.xtext.xbase.ide/src-gen/org/eclipse/xtext/xbase/annotations/ide/AbstractXbaseWithAnnotationsIdeModule.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2010, 2017 itemis AG (http://www.itemis.eu) and others.
+ * Copyright (c) 2010, 2018 itemis AG (http://www.itemis.eu) and others.
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase.ide/src-gen/org/eclipse/xtext/xbase/annotations/ide/AbstractXbaseWithAnnotationsIdeModule.java]: oudated

**Comment by cdietrich** [comment]
@szarnekow is this obsolete an can be closed?
